Okta.AspNetCore/OktaMvcOptions.cs

```diff
@@ -3,22 +3,61 @@
 // Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
 // </copyright>
 
+using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
 
 namespace Okta.AspNetCore
 {
     public class OktaMvcOptions : AspNet.Abstractions.OktaWebOptions
     {
+        /// <summary>
+        /// Gets or sets the client secret of your Okta Application.
+        /// </summary>
+        /// <value>The client secret.</value>
         public string ClientSecret { get; set; }
 
+        /// <summary>
+        /// Gets or sets the client ID of your Okta Application.
+        /// </summary>
+        /// <value>The client ID.</value>
         public string ClientId { get; set; }
 
+        /// <summary>
+        /// Gets or sets the location Okta should redirect to process a login.
+        /// </summary>
+        /// <value>The callback path.</value>
         public string CallbackPath { get; set; } = OktaDefaults.CallbackPath;
 
+        /// <summary>
+        /// Gets or sets the location Okta should redirect to after logout. If blank, Okta will redirect to the Okta login page.
+        /// </summary>
+        /// <value>The post-logout redirect URI.</value>
         public string PostLogoutRedirectUri { get; set; }
 
+        /// <summary>
+        /// Gets or sets the OAuth 2.0/OpenID Connect scopes to request when logging in.
+        /// </summary>
+        /// <value>The scope.</value>
         public IList<string> Scope { get; set; } = OktaDefaults.Scope;
 
-        public bool GetClaimsFromUserInfoEndpoint { get; set; } = false;
+        /// <summary>
+        /// Gets or sets a value indicating whether to retrieve additional claims from the UserInfo endpoint after login.
+        /// </summary>
+        /// <value>The GetClaimsFromUserInfoEndpoint flag.</value>
+        public bool GetClaimsFromUserInfoEndpoint { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the event invoked when an IdToken has been validated and produced an AuthenticationTicket.
+        /// </summary>
+        /// <value>The OnTokenValidated event.</value>
+        public Func<TokenValidatedContext, Task> OnTokenValidated { get; set; } = context => Task.CompletedTask;
```

Comment:
This code looks weird... I get what it's doing, but it feels unlike any C# code I've seen.  In my experience this would be implemented as an event (which is null by default), and then the caller would check if the event was null before calling it.  But instead you have a `Func` with a default value that returns a completed `Task`.  Is this a common pattern in .Net Core these days?  It feels weird, but if you think it's the right way to go then I'll trust your judgement.

Reply:
This is actually a common pattern in the OIDC implementation: https://github.com/aspnet/Security/blob/master/src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs.

Maybe the null checking is not completely necessary, I'm gonna double check that to see what's the behavior when users set this event to null.

Reply:
I confirm that the null checking is necessary.

Reply:
The null checking is mainly for the pattern where we are using events, since if there’s no event handler then the event is bull so you don’t want to call it directly. In this case you are providing a default implementation, so it shouldn’t be null unless the user sets it to null specifically.  That doesn’t feel like something that’s likely to happen, and if it does happen it’s kind of on the user to deal with it. So I don’t believe we actually need null checking in this case.